comp/core/workloadmeta/collectors/util/ecs_util.go

```diff
@@ -265,3 +266,18 @@ func parseClusterName(cluster string) string {
 	}
 	return parts[1]
 }
+
+// ParseECSAgentVersion parses the ECS agent version from the version string
+// Instance metadata returns the version in the format `Amazon ECS Agent - v1.30.0 (02ff320c)`
+func ParseECSAgentVersion(s string) string {
+	// \d+\.\d+\.\d+ for versions like 1.32.0
+	// (-\w+)? for optional pre-release tags like -beta
+	// |\d+\.\d+ for shorter versions like 0.1
+	// |\d+ for versions like 1
+	re := regexp.MustCompile(`v(\d+\.\d+\.\d+(-\w+)?|\d+\.\d+(-\w+)?|\d+(-\w+)?)`)
```

Comment:
Constant regex compilation should be done outside the function so that it is done once at program startup instead of at each function call.